Apps/Sandcastle/gallery/development/3D Models Instancing.html

```diff
@@ -0,0 +1,155 @@
+</head>
+<body class="sandcastle-loading" data-sandcastle-bucket="bucket-requirejs.html">
+<style>
+    @import url(../templates/bucket.css);
+</style>
+<div id="cesiumContainer" class="fullSize"></div>
+<div id="loadingOverlay"><h1>Loading...</h1></div>
+<div id="toolbar"></div>
+<script id="cesium_sandcastle_script">
+function startup(Cesium) {
+    "use strict";
+//Sandcastle_Begin
+
+var debugShowBoundingVolume = false;
+var debugWireframe = false;
+var globeVisible = false;
+
+var viewer = new Cesium.Viewer('cesiumContainer', {
+    globe : globeVisible,
+    skyAtmosphere : globeVisible
+});
+
+var scene = viewer.scene;
+
+function createCollection(url, count) {
+    var instances = [];
+    var longitude = -123.0744619;
+    var latitude = 44.0503706;
+    var height = 5000.0;
+    for (var i = 0; i < count; ++i) {
+        var heading = Math.random();
+        var pitch = Math.random();
+        var roll = Math.random();
+        var position = new Cesium.Cartesian3.fromDegrees(longitude, latitude, height + i * 20);
+
+        var modelMatrix = Cesium.Transforms.headingPitchRollToFixedFrame(position, heading, pitch, roll);
+        instances.push({
+            modelMatrix : modelMatrix
+        });
+    }
+
+    var collection = scene.primitives.add(new Cesium.ModelInstanceCollection({
+        url : url,
+        instances : instances,
+        debugShowBoundingVolume : debugShowBoundingVolume,
+        debugWireframe : debugWireframe
+    }));
+
+    collection.readyPromise.then(function(collection) {
+        var model = collection.model;
+
+        // Play and loop all animations at half-speed
+        model.activeAnimations.addAll({
+            speedup : 0.5,
+            loop : Cesium.ModelAnimationLoop.REPEAT
+        });
+
+        var camera = viewer.camera;
+
+        // Zoom to model
+        var controller = scene.screenSpaceCameraController;
+        var r = 2.0 * Math.max(model.boundingSphere.radius, camera.frustum.near);
+        controller.minimumZoomDistance = r * 0.5;
+
+        var origin = collection.boundingSphere.center;
+        var heading = Cesium.Math.toRadians(230.0);
+        var pitch = Cesium.Math.toRadians(-20.0);
+        camera.lookAt(origin, new Cesium.HeadingPitchRange(heading, pitch, r * 2.0));
+    }).otherwise(function(error){
+        window.alert(error);
+    });
+}
+
+var handler = new Cesium.ScreenSpaceEventHandler(scene.canvas);
+handler.setInputAction(function(movement) {
+    var pick = scene.pick(movement.endPosition);
+    if (Cesium.defined(pick) && Cesium.defined(pick.node) && Cesium.defined(pick.mesh)) {
+        // Output glTF node and mesh under the mouse. 
+        console.log('node: ' + pick.node.name + '. mesh: ' + pick.mesh.name);
+    }
+}, Cesium.ScreenSpaceEventType.MOUSE_MOVE);
+
+///////////////////////////////////////////////////////////////////////////
+
+createCollection('../../SampleData/models/CesiumAir/Cesium_Air.bgltf', 10);
+
+//function createModel(url, height) {
```

Comment:
I use this code for testing sometimes, and I will delete it once I'm further along with testing. In fact I'll probably need it back for the performance testing.